public/app/plugins/panel/canvas/CanvasPanel.tsx

```diff
@@ -73,14 +93,27 @@ export class CanvasPanel extends Component<Props, State> {
               selected: v,
               layer: this.scene.root,
             });
+
+            activeCanvasPanel = this;
+            activePanelSubject.next({ panel: this });
+
+            canvasInstances.forEach((canvasInstance) => {
+              if (canvasInstance !== activeCanvasPanel) {
+                canvasInstance.scene.clearCurrentSelection(true);
+              }
+            });
           },
         })
       );
     }
+
+    canvasInstances.push(this);
   }
 
   componentWillUnmount() {
     this.subs.unsubscribe();
+    isInlineEditOpen = false;
+    canvasInstances = [];
```

Comment:
Do we want to clear all canvasInstances when one panel unmounts or just remove that panel from the array?

Reply:
Updated. I had a wrong view on when the unmount actually happens.